backup/data.go

```diff
@@ -86,8 +86,20 @@ func CopyTableOut(connectionPool *dbconn.DBConn, table Table, destinationToWrite
 		// process column names to exclude generated columns from data copy out
 		columnNames = ConstructTableAttributesList(table.ColumnDefs)
 	}
+	tableName := table.FQN()
+	ignoreExternalPartitions := "IGNORE EXTERNAL PARTITIONS"
+	if table.ExtensionTableConfig != nil && *table.ExtensionTableConfig != "" {
+		if columnNames == "" {
+			columnNames = ConstructTableAttributesList(table.ColumnDefs)
+		}
+		tableName = fmt.Sprintf("(SELECT %s FROM %s %s)", columnNames[1:len(columnNames)-1], tableName, *table.ExtensionTableConfig)
+		columnNames = ""
+		if connectionPool.Version.Before("7") {
+			ignoreExternalPartitions = ""
+		}
+	}
 
-	query := fmt.Sprintf("COPY %s%s TO %s WITH CSV DELIMITER '%s' ON SEGMENT IGNORE EXTERNAL PARTITIONS;", table.FQN(), columnNames, copyCommand, tableDelim)
+	query := fmt.Sprintf("COPY %s%s TO %s WITH CSV DELIMITER '%s' ON SEGMENT %s;", tableName, columnNames, copyCommand, tableDelim, ignoreExternalPartitions)
```

Comment:
I would move the last space to `ignoreExternalPartitions`, not to print it if `ignoreExternalPartitions` is empty.

Reply:
modev